crates/rpc/src/lib.rs

```diff
@@ -18,6 +20,31 @@ impl BeerusRpc {
     }
 
     pub async fn run(self) -> Result<(SocketAddr, ServerHandle), Error> {
+        let local_spec_version =
+            self.spec_version().await.expect("Should never happen");
+
+        let remote_spec_version = self
+            .beerus
+            .starknet_client
+            .spec_version()
+            .await;
+
+        // check remote spec version
+        match remote_spec_version {
+            Ok(remote_version) => {
+                if remote_version != local_spec_version {
+                    error!(
+                        "Spec version mismatch between Beerus {} and remote Starknet RPC {}", local_spec_version, remote_version
+                    );
+                    return Err(Error::Custom(
+                        format!("Spec version mismatch between Beerus {} and remote Starkent RPC {}", local_spec_version, remote_version)
+                            .to_string(),
+                    ));
+                }
+            },
+            Err(_) => return Err(Error::Custom("Failed to fetch remote spec version".to_string())),
```

Comment:
Why drop the cause of the error here, and not just fail fast when call to `spec_version` fails?

Reply:
`run` function is returning `jsonrpsee::Error` and error returned from call is `starknet::providers::ProviderError`
One way we can do it is implement another BeerusRPCError from trait in error.rs for `jsonrpsee::Error` and then use BeerusRPCError in return in `run` function.
If this sounds good, I can implement this or if there is another way to resolve it, I can do it that way.

Reply:
Adding extra variant to `BeerusRpcError` for `jsonrpsee::Error` seems like a reasonable way to proceed. WDYT @pierre-l ?

Reply:
To make this approach work, we will need to implement `std::error::Error` which in turn required implementation of `std::fmt::Display` for BeerusRpcError. If we are going with this approach, any pointers on implementing these traits would be helpful

Reply:
[thiserror](https://crates.io/crates/thiserror)